src/tests/functional/shared_test_classes/src/single_op/non_max_suppression.cpp

```diff
@@ -148,5 +153,248 @@ void Nms9LayerTest::SetUp() {
     function = std::make_shared<ov::Model>(nms, params, "NMS");
 }
 
+void NmsLayerTest::compare(const std::vector<ov::Tensor>& expected, const std::vector<ov::Tensor>& actual) {
+    CompareBBoxes(expected, actual);
+}
+
+namespace {
+
+typedef struct Rect {
+    int32_t x1;
+    int32_t y1;
+    int32_t x2;
+    int32_t y2;
+} Rect;
+
+class Box {
+public:
+    Box() = default;
+
+    Box(int32_t batch_id, int32_t class_id, int32_t box_id, Rect rect, float score) {
+        this->batch_id = batch_id;
+        this->class_id = class_id;
+        this->box_id = box_id;
+        this->rect = rect;
+        this->score = score;
+    }
+
+    int32_t batch_id;
+    int32_t class_id;
+    int32_t box_id;
+    Rect rect;
+    float score;
+};
+
+template <typename fromT, typename toT>
+void convert(fromT* src, toT* dst, size_t size) {
+    std::transform(src, src + size, dst, [](fromT el) -> toT {
+        return static_cast<toT>(el);
+    });
+}
+
+}  // namespace
+
+// 1: selected_indices - tensor of type T_IND and shape [number of selected boxes, 3] containing information about
+// selected boxes as triplets [batch_index, class_index, box_index]. 2: selected_scores - tensor of type T_THRESHOLDS
+// and shape [number of selected boxes, 3] containing information about scores for each selected box as triplets
+//    [batch_index, class_index, box_score].
+// 3: valid_outputs - 1D tensor with 1 element of type T_IND representing the total number of selected boxes.
+void NmsLayerTest::CompareBBoxes(const std::vector<ov::Tensor>& expected, const std::vector<ov::Tensor>& actual) {
```

Comment:
Let's rename in snake_case style

Reply:
All the code of `CompareBBoxes` is 90% a duplication of src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/non_max_suppression.cpp and src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/non_max_suppression.cpp. Let's try to combine them all in one place

Reply:
done